q/compare.go

```diff
@@ -88,7 +88,20 @@ func compare(a, b interface{}, tok token.Token) bool {
 		}
 	}
 
-	if reflect.TypeOf(a).String() == "time.Time" && reflect.TypeOf(b).String() == "time.Time" {
+	if reflect.TypeOf(a) != nil && (reflect.TypeOf(a).String() == "time.Time" || reflect.TypeOf(a).String() == "*time.Time") &&
```

Comment:
I think we can avoid the multiple calls to `reflect.TypeOf` by caching them in a variable before the `if`, what do you think?

Reply:
You are right, I've made the changes.